architecture/readme.md

```diff
@@ -0,0 +1,53 @@
+<!-- This is a generated file. Use `./gradlew :architectureDoc` to generate -->
```

Comment:
If there was a way to generate this diagram and include it in `bt-internal-docs` this would be terrific.  Or perhaps it belongs in the public Gradle docs?

Reply:
>  It's absolutely a markdown file.

Are you sure?  The Mermaid site makes a point to say it's "markdown-inspired".  It's not really "plain vanilla" markdown in any case, and I think if you're viewing the file with a markdown tool that doesn't speak Mermaid, you might have some trouble.

Mermaid recommends `.mermaid` or `.mmd` [here](https://mermaid.js.org/ecosystem/integrations-create.html#:~:text=File%20Extension%20%E2%80%8B,mmd%20file%20extensions.).

Reply:
It's a markdown file with some markdown content, such as the header and a mermaid diagram enclosed in a markdown block. It will contain some words and other content in the future. Github renders it just fine, which is all we need.